pandas/core/groupby.py

```diff
@@ -2315,6 +2315,11 @@ def __init__(self, index, grouper=None, obj=None, name=None, level=None,
                 # groupby
                 else:
                     cat = self.grouper.unique()
```

Comment:
hmm I am wondering why we are doing any of this. isn't ``self.grouper.categories`` the end result that we want anyway?